src/pages/Lending/Pool/components/Repay/RepayInput.tsx

```diff
@@ -208,6 +211,65 @@ export const RepayInput = ({
     return bnOrZero(repaymentAmountFiatUserCurrency).div(repaymentAssetMarketData.price).toFixed()
   }, [repaymentAmountFiatUserCurrency, repaymentAssetMarketData.price])
 
+  const { data: estimatedFeesData, isLoading: isEstimatedFeesDataLoading } =
+    useQuoteEstimatedFeesQuery({
+      collateralAssetId,
+      collateralAccountId,
+      repaymentAccountId,
+      repaymentPercent,
+      repaymentAsset,
+    })
+
+  const balanceFilter = useMemo(
+    () => ({ assetId: repaymentAsset?.assetId ?? '', accountId: repaymentAccountId }),
+    [repaymentAsset?.assetId, repaymentAccountId],
+  )
+
+  const balance = useAppSelector(state =>
+    selectPortfolioCryptoBalanceBaseUnitByFilter(state, balanceFilter),
+  )
+  const amountAvailableCryptoPrecision = useMemo(
+    () => bnOrZero(balance).div(bn(10).pow(collateralAsset?.precision ?? '0')),
+    [balance, collateralAsset?.precision],
+  )
+
+  const hasEnoughBalance = useMemo(
+    () =>
+      bnOrZero(repaymentAmountCryptoPrecision)
+        .plus(
+          bnOrZero(estimatedFeesData?.txFeeCryptoBaseUnit).div(
+            bn(10).pow(repaymentAsset?.precision ?? '0'),
+          ),
+        )
+        .lte(amountAvailableCryptoPrecision),
+    [
+      amountAvailableCryptoPrecision,
+      estimatedFeesData?.txFeeCryptoBaseUnit,
+      repaymentAmountCryptoPrecision,
+      repaymentAsset?.precision,
+    ],
+  )
+
+  const quoteErrorTranslation = useMemo(() => {
+    if (!hasEnoughBalance) return 'common.insufficientFunds'
+    console.log({ lendingQuoteCloseError })
+    if (isLendingQuoteCloseError) {
+      if (
+        /not enough fee/.test(lendingQuoteCloseError.message) ||
+        /not enough to pay transaction fee/.test(lendingQuoteCloseError.message)
+      ) {
+        return 'trade.errors.amountTooSmallUnknownMinimum'
+      }
+      if (
+        /loan hasn't reached maturity/.test(lendingQuoteCloseError.message) ||
+        /loan repayment is unavailable/.test(lendingQuoteCloseError.message)
```

Comment:
any value in ignoring case?

Reply:
why not 🧠 d9507a34067cdb1221f5228a17e6b4b704a2ab9c